src/zenml/config/step_configurations.py

```diff
@@ -145,8 +145,8 @@ class StepConfigurationUpdate(StrictBaseModel):
     enable_artifact_metadata: Optional[bool] = None
     enable_artifact_visualization: Optional[bool] = None
     enable_step_logs: Optional[bool] = None
-    step_operator: Optional[str] = None
-    experiment_tracker: Optional[str] = None
+    step_operator: Optional[Union[bool, str]] = None
```

Comment:
Are we planning to remove the `str` option in the future? 

If so, should we add a deprecation warning somewhere if people still use it?

Reply:
On the other hand, if we allow multiple instances of the same type of component in the future, it might make sense to keep the `str` option as well.